File: dsa-practice-ram/dsa-practice-ram/Arrays/SortZeosOnes.cs
namespace dsa_practice_ram.Arrays
{
    public class SortZeosOnes
    {
        public void SortZeosOness()
        {
            int[] array = { 1, 0, 0, 1, 1, 1, 0, 0, 1 };
            int zerosTotal = 0;

            foreach (var item in array)
            {
                if (item == 0)
                    zerosTotal++;
            }

            for (int i = 0; i < zerosTotal; i++)
            {
                array[i] = 0;
            }
            for (int i = zerosTotal; i < array.Length; i++)
            {
                array[i] = 1;
            }

            Console.WriteLine(string.Join(", ", array));
        }
    }
}

File: MyLearning/PythonPractice/dsa_practice/arrays/sort_zeos_ones.py
"""Sort zeros and ones."""

# C# SortZeosOnes.SortZeosOness

def sort_zeos_oness(array=None):
    """Sort an array containing 0s and 1s by counting zeros.

    Problem: Move all 0s to the front and 1s to the end.
    Approach: Count zeros, then overwrite positions accordingly.
    Time complexity: O(n)
    Space complexity: O(1) extra (in-place on a local list)
    """
    if array is None:
        array = [1, 0, 0, 1, 1, 1, 0, 0, 1]

    nums = list(array)
    zeros_total = 0

    for item in nums:
        if item == 0:
            zeros_total += 1

    for i in range(zeros_total):
        nums[i] = 0
    for i in range(zeros_total, len(nums)):
        nums[i] = 1

    return nums
